packages/compiler/src/core/parser.ts

```diff
@@ -2425,7 +2425,20 @@ function createParser(code: string | SourceFile, options: ParseOptions = {}): Pa
     messageId: keyof CompilerDiagnostics["doc-invalid-identifier"]
   ): ParamLikeTag {
     const name = parseDocIdentifier(messageId);
+
+    if (messageId === "param") {
+      // The TSDoc spec expects a hyphen in the `@param` case, but it does
+      // not require it in practice. Thus, TypeSpec should handle both cases.
+      while (parseOptional(Token.Whitespace));
+      if (token() === Token.Hyphen) {
+        // The doc content starts with a hyphen, so skip it and subsequent whitespace
+        nextToken();
```

Comment:
```suggestion
      while (parseOptional(Token.Whitespace));
      if (parseOptional(Token.Hyphen)) {
        // The doc content starts with a hyphen, so skip it and subsequent whitespace
```
I think you can use  `if(parseOptional(Token.Hyphen)` instead and skip calling nextToken below